stdlib/public/core/KeyPath.swift

```diff
@@ -2448,7 +2448,6 @@ public func _swift_getKeyPath(pattern: UnsafeMutableRawPointer,
   // These are resolved dynamically, so that they always reflect the dynamic
   // capability of the properties involved.
 
-  let oncePtrPtr = pattern
```

Comment:
This is an example of [what I discussed in this reply](
https://github.com/apple/swift/pull/36134#issuecomment-786626319) as a legitimate use case for using a different name while binding to the same value.

Here, this binding stresses the relationship between `oncePtrPtr` and `oncePtrOffset` which is not clear for the reader at the point of use when the name of the binding is `pattern`.

Reply:
Okay, thanks for bringing this up again. I'll echo this note in #39179.